jslib/core/http/Client.ts

```diff
@@ -0,0 +1,105 @@
+declare const loris: any;
+import { Query, QueryParam } from './Query';
+import { Errors } from '../';
+
+export interface ErrorContext {
+  key: string | number; // The key that triggered the custom message (e.g., 'ApiNetworkError' or 404)
+  request: Request,
+  response?: Response,
+}
+
+export class Client<T> {
+  protected baseUrl: string;
+  protected subEndpoint?: string;
+  public getCustomMessage: (
```

Comment:
this is to allow each Client concrete class to define it's own errors across all the error types, not sure if it's the best implementation, but it seems to work